code/modules/fabrication/fabricator_books.dm

```diff
@@ -14,8 +14,13 @@
 	)
 	color_selectable = TRUE
 
-/obj/machinery/fabricator/book/do_build(datum/fabricator_recipe/recipe, amount)
-	. = recipe.build(get_turf(src), amount, selected_color)
+/obj/machinery/fabricator/book/make_order(datum/fabricator_recipe/recipe, multiplier)
+	var/datum/fabricator_build_order/order = ..()
+	LAZYSET(order.data, "selected_color", selected_color)
+	return order
+
+/obj/machinery/fabricator/book/do_build(datum/fabricator_build_order/order)
+	. = order.target_recipe.build(get_turf(src), order.multiplier, LAZYACCESS(order.data, "selected_color"))
```

Comment:
I think at the minimum this should be like `order.build(blah)`, and then `/order/proc/build(blah){target_recipe.build(blah)}`. Likely the superior pattern would be something like `target_recipe.build(order){looks for appropriate metadata on order}`, but may be out of scope.

Reply:
Yeah, I thought that as well, but adding a proc that just calls another proc just rubs me the wrong way. 
Also, technically, an order is metadata for queued recipes already. So dunno what you mean about looking for metadata on order?